\section{Material}
Next generation sequencing (NGS) and magnetic resonance imaging (MRI) data were obtained from Alzheimerâ€™s Disease Neuroimaging Initiative (ADNI). A totol of 808 subjects at the screening and baseline of ADNI1 and ADNI2 study have both types of profiles available, alongside with disease diagnosis, demographics, and the genotype of APOE $\epsilon$4.

The image preprocessing is done by the \FS suite freely distributed online (\url{http://surfer.nmr.mgh.harvard.edu}) \cite{FS:Intro}. A variant is then defined as a vertex in the reconstructed surface, and its value being the thickness of cerabral gray matter at that vertex. 

The structure MRI is data first went throught a series of preprocessing including special registration, skull stripping, cortical/subcortical segmentation, white/gray matter segregation, vexel intensity normalization, reconstruction of cortical surface, surface registration, and surface paceration. The entire pipline is implemented by \FS - a neuroimage analysis package developed by Fisher and Dale et.al. \cite{FS:Intro}, and currently maintained by \textit{the Laboratory for Computational Neuroimaging (LCN)}  at \textit {the Athinoula A. Martinos Center for Biomedical Imaging}. \FS is freely distributed online (\url{http://surfer.nmr.mgh.harvard.edu}). For each cerebral hemisphere, the cortical surface reconstructs from the MRI slices is spaned by $10\times2^16 = 163840$ vertices in the 3D space, connected by triangles. Each vertex is then treated as a variant, with a number of geometrical attribute calculated during the surface reconstruction attatched to it, such as the coordinate of the vertex, the gray matter thickness, average curvature, local area and volume around its vicinity. Of all these attributes bounded with each vertex, that is, a image variant, currently we took the gray matter thickness as its value. The last preprocessing step of \FS-- surface paceration, would divide the entire cortical surface into 68 anatomical regions of interest (ROI), 34 for each hemisphere. For real data analysis, the 68 ROI are treated as our testing units, for simulation study, small ovals of $512$ vertices (mean diametter=28mm) were randomly picked from either hemisphere surface for each iteration.

Rigorous quality control had been done by ADNI during variant calling process, thus the WGS data from ADNI do not require intensive preprocessing. For genomic profile, the testing unit for both real data and simulation study are based on gene. The chromosome location of known genes were queried from the table of genomic features of human reference genome assemble version 38, maintained by Genome Reference Consortium (GRCh38). An extra 5k flanking basepairs were attached to both ends of a gene when searching for vairants in a testing unit. Despit the added flanking region, some unit contains no genomic variant, and they were excluded from further opertations.