\section{Discussion}
We applied the recent machine learning trend, the ``deep learning'' to build stacked autoencoders (SAs) for our cortex region and abstract high order features from them. These high order features are not only lower in dimensionality, but also boosted statistical power slightly wherever imaging profile is involved. Also, the unsupervised training means better chance of getting training materials. In our case, materials that excluded from association analysis due to uncertain diagnosis still helped building the SAs. In the future, imaging data from alternative sources can serve as training material to make existing SAs more ``knowledgeable'' to improve encoding quality, even if the data was not collected for our design. A good example is ADNI's nearly 1,800 baseline participants, aside from $806$ participants included for having both type of profiles, those 1,000 left without NGS data can still contribute to the refinement of SAs.

The proposed method enhanced the detection power of genomic association analysis with additional information from imaging profile, which is also a robust method in case the imaging bio-markers neither directly associating with nor mediating the health outcome. A immediate expansion is to incorporate other ``omics'' with proper U kernels, which will aid future association analysis. Some common candidates are transcriptomics which is close to the upstream genomics in the casual pathway, and the symbiotic microbiome that complements human genome.

We also showed that grouping and signal aggregation is helpful to imaging variants of continuous values, knowing the vertices closely located in a cortex region are highly connected. The strategy can be easily applied to other profile types, for MRI slices one could analyze voxels located in the same sub-cortical region defined by established functional anatomy, for transcriptomics, the grouping of RNA can refer to known biological pathways or results of clustering analysis. 

Although the study objectives is met, the potent of artificial neural network (ANN) is only explored superficially. If the computation can be made more efficient, the SAs should go deeper, that is, larger number of layers, more compact output, and preferably lower reconstruction loss. A deeper SA is more difficult to train, but we believe the smaller the code it produces, the more meaningful the features it abstracts, which means a more pronounced boost to the subsequent statistical analysis. Also, the simulation may not adequately demonstrate the benefit of high order features over raw imaging data, as we known the deep ANN is imitating visual processing that focuses on major cortical features such as the dozens of gyri and sulci that is recognizable by naked eyes, but the simulated effects were based on 5\% of the variants scattered all over the cortex, not bounded with large, obvious features. Lastly, we would like to build SAs for genomic regions as well, but major improvement in computation is needed because of the large number of genes, even though 98\% of them have fewer variants than an average cortex region.

The similarity U statistics is not without limit either, although it can easily take on multiple components, it can not disentangle the effect composition but instead only tells if there exists association among some of the components, with or without some unknown type of interaction. In a customized study with multiple high dimensional profiles, one may still employ a linear model which explicitly models the effect size of each component and the their interactions.

As for signal aggregation, deciding the grouping criteria is always an issue, and sometimes a pain for being highly arbitrary or haveing little to no prior knowledge to refer to. The biological pathways may also overlap so the same variant may appear in more than one group. Beside, signal aggregation is, after all, a compromise between accuracy and efficiency, should adequate sample is available, one can still rely on a per-variant screening procedure which not only precisely locate the significant variants, but also save the effort of deciding for a grouping scheme. In the future, we may favor MRI slice over 3D cortex, even if the later has richer information (e.g.\ coordinates, thickness, curvature, etc.),  because voxels in the MRI can be easily divided into equal sized volumes satisfying any resolution of desire, which balances the accuracy and power without consulting any prior knowledge.

