
\documentclass{article}
\begin{document}
\textbf{Abstract}
The rapid accumulation of conprehensive whole genome sequence(WGS) and brain image data(MRI) mandates the development of analysical procedures capable of utilizing both genetic and image information to detect predicative biomarkers of both type for CNS diseases of interest such as Alzheimer. Both type of profiles however, poses "the curse of dimensionality" due to the intrisic colossal size inherited from the vast wealth of genomic variants and brain surface vertices. In this work we tackled the dimensionality issue on the image side by piping the MRI structure data of one anatomycal region through an stacked denosing autoencoder(SDA) constructed using deep learning approach, reducing the sample features from millions to 10; For the the whole genome data, an IBS kernel were used to aggregate rare variants by gene units, which reduces the total number of tests/variants and at the same time boost statistical power. With the condensed profiles of lower dimension, we were able to perform a joint association test against disease phenotype with a weighted U statistics which bypass the assumptions on distribution and mediation effect. Preliminary simulation showed correct type I error rate and moderate power boost incompression with tests using only genetic profile or non-surface vertices based image data(e.g. regional volumn)
\end{document}
