\documentclass[twocolumn]{article}

\usepackage{syntonly}
%\syntaxonly
\usepackage{amsmath}

\pagestyle{headings}

\author{Xiaoran Tong}

\begin{document}
\title{A non-parametric method for joint association analysis of sequencing and imaging data}
\maketitle

\begin{abstract}
The rapid development of next generation genome sequencing and neroimaging technology, and the consequental reduction in sampling cost, facilitated the establishment of large, mult-site cohort maintaining both type of data. Such growing wealth of whole genome sequencing (WGS) data and magnetic resonance imaging (MRI) data, mandates the response analytical methods capable of utilizing both type of information to identify predictive biomarkers associated with neurological disorders. Such attemp, however, are met with "the curse of dimensionality", due to the large number of variants in the genome and image themself or products of preprocessing. In this work, we tackled the dimensionality of 3D brain surface by training a stacked denoising autoencoder (SDA) with the deep learning algorithm. A weighted U statistic is then used to evaluate the joint association of vertex and genome data with the phenotype. We showed by simulation that the method maintains the correct type 1 error rate, and achieved a statistical power higher then using either genome or vertex data alone, or methods relying on exhaustive per-elemet test. To illustrate our approach, we apply the proposed method to the genomic sequencing and neuroimage data from the Alzheimer's disease Neuroimaging Initiative (ADNI).
\end{abstract}

\section{Introduction}
We know the voxels/vertices near by are closely related due to the fact that they represent physical connected brain tissues, and the location of the vertices and the gray matter thickness at that location are two distinct type of information.
To speed up learning and improve generalization we should build a priori knowledge into the network and let it use the information in the training data to discover structure that we do not already understand or difficault to modulate by human labor.

6.9. Self-supervised backpropagation
One drawback of the standard form of backpropagation is that it requires an external supervisor to specify the desired states of the output units (or a transformed "image" of the desired states). It can be converted into an unsupervised procedure by using the input itself to do the supervision, using a multi-layer "encoder" network [2] in which the desired output vector is identical with the input vector. The network must learn to compute an approximation to the identity mapping for all the input vectors in its training set, and if the middle layer of the network contains fewer units than the input
layer, the learning procedure must construct a compact, invertible code for each input vector. This code can then be used as the input to later stages of processing.
The use of self-supervised backpropagation to construct compact codes resembles the use of principal components analysis to perform dimensionality reduction, but it has the advantage that it allows the code to be a nonlinear transform of the input vector. This form of backpropagation has been used successfully to compress images [19] and to compress speech waves [25]. A variation of it has been used to extract the underlying degrees of freedom of simple shapes [83].

\section{Material}
Whole genome sequencing (WGS) and magnetic resonance imaging (MRI) data were obtained from Alzheimerâ€™s Disease Neuroimaging Initiative (ADNI). A totol of 808 subjects at the screening and baseline of ADNI1 and ADNI2 study have both types of profiles sampled, alongside with Alzheimer Disease diagnosis, demographics, and the genotype of APOE $\epsilon$4.

The MRI data first went throught a series of preprocessing pipeline composed of special registration, skull stripping, cortical/subcortical segmentation, white/gray matter segregation, vexel intensity normalization, reconstruction of brain surface, surface registration and virtual anatomical paceration. The whole pipeline is supported by \emph{FreeSurfer} - a software initiated by [Dale and Fischl et.al.] and currently maintained by \textit{the Laboratory for Computational Neuroimaging (LCN)}  at \textit {the Athinoula A. Martinos Center for Biomedical Imaging}. The preprocessing produces a 3D brain surface, with each hamersphere expressed by $10\times2^16$ vertices connected by triangles, and vitually segemented into 34 anatomical regions [pic ?]. For each vertex, besides its location in the 3D space, statistics ascribing the tissure at its close vicility were also calculated, among which the most important properties are gray matter thickness and sulcity (positive means the vertex is in a gyrus, otherwise in a sulcus). For real data analysis, the 68 anatomical regions are treated as testing unit; for a single iteration in a simulation study, small ovals of $512=2^9$ vertices (mean diametter=28mm) were randomly picked from the whole surface.

Rigorous quality control had been done by ADNI during variant calling process, thus the WGS data from ADNI do not require intensive preprocessing. For genomic profile, the testing unit for both real data and simulation study are based on gene. The chromosome location of known genes were queried from the table of genomic features of human reference genome assemble version 38, maintained by Genome Reference Consortium (GRCh38). An extra 5k flanking basepairs were attached to both ends of a gene when searching for vairants in a testing unit. Despit the added flanking region, some unit contains no genomic variant, and they were excluded from further opertations.

\section{Method}

To simoteniously detect association among genetics, neuroimaging and phenotype profiles, we implemented a generalized multivariate similarity U statistic tesd postulated by [changshuai et. al.]. The U statistic is derived by summerizing the product of pairwise similarity between subjects, with respect to the three aforementioned profile types. Of three profiles, the brain surface reconstructed from MRI scan has the most numerous elements, in that the 68 anatomical region in are comprised of few hundreds to more than ten thousands of vertices. The deep-learning algorithm, developed for computer vision, is introduced. A a 5-layer stacked denosing autoencoder (SDA) was trained with the vertex data until the encoding loss reaches a minimum, the vertex data were then pumped throught the converged SDA for a condensed code, resulting in a 32 fold reduction of dimenstionality, and the extrapolation of high order features of the original brain surface.  by a gaussian kernal funciton to aggregate the code. In a similar manner, we used a identity-by-state (IBS) kernel to aggregate information over sequencing variants by gene unit. 

  To derive the proposed U statistic, one U-kernel function and one or more U-weight functions must be constructed. These functions are centered measurements of pairwise subject similarity with respect to their profiles. The measurement funciton can be flexable depending on the type of profile and the hypothesis in mind, as long as the function is symmetric and has finite second moment. Thus, centralized function $f$ is a valid U-weight or U-kernel if $f(x_i,x_j)=f(x_j,x_i)$ and $E(f^2(X_1, X_2))<+\infty$ are satisfied. The current study has three types profile, namely the genetics, vertex code, and the phenotype, for which the three pairwise similarity measurement were chosen according to common practice, respectively.

For biallelic genetic variants whose value was taken from minor allele count ${0, 1, 2}$, a common choice is the taking the complement of Manhattan Distance (wMD):
\begin{equation} \label{eq_wSG}
\begin{split}
  S_{ij}^G &= wMD(g_{i.},g_{j.}) \\
  &= \frac{\sum_{m=1}^{|G|}{w_m(2-|g_{im}-g_{jm}|)}} {2\sum_{m=1}^{|G|}{w_m}},
\end{split}
\end{equation}
where $g.m$ is the genotype of the $m th.$ variant (e.g. a SNP, indel, or deletion, ect.) in a testing unit (e.g. a gene); $w_m$ is the weight assigned to the $m th.$ variant depending on \textit{a prior} hypothesis, one common example is the reversed square root of minor allele frequency which emphasizes more on the rare variants, corresponding to the rare variant large effect hypothesis:
\begin{displaymath}w_m=\frac{1}{\sqrt{MAF(g_m)(1-MAF(g_m))}}.\end{displaymath}
Without any \textit{a prior} hypothesis of the relative importance of genomic variants, an unweighted complement of Manhatten Distance (uMD) can be used by forcing $w_m \equiv 1$:
\begin{equation*} \label{eq_uSG}
\begin{split}
  S_{ij}^G &= uMD(g_{i.}, g_{j.}) \\
  &= \frac{\sum_{m = 1}^{|G|}{(2-|g_{im} - g_{jm}|)}}  {2|G|}.
\end{split}
\end{equation*}

The encoded surface is another type of profile, since the code are real values between $[0,1]$, one could exploit the weighted Euclidian Distance (wED) to build a similarity measurement:
\begin{equation} \label{eq_wSV}
\begin{split}
  S_{ij}^V &= wED(v_{i.},v_{j.}) \\
  &= \exp
  {
    \Big[-\frac{\sum_{m=1}^{|V|}{w_m(u_{im}-u_{jm})^2}} {\sum_{m=1}^{|V|}{w_m}}\Big]
  },
\end{split}
\end{equation}
where $v_{im} and v_{jm}$ are values of the $m th.$ dimension of the surface vertices code for a pair of subject, $w_m$ is the weight assigned to that dimension, and $|V|$ is the total dimension of the code. Because there is no knowledge regarding the relative importance for each of the $|V|$ code components, the measurement is simplifed by equal weighting:
\begin{equation*} \label{eq_uSV}
\begin{split}
  S_{ij}^V &= uED(v_{i.},v_{j.}) \\
  &= \exp
  {
    \Big[-\frac{\sum_{m=1}^{|V|}{(u_{im}-u_{jm})^2}} {|V|}\Big]
  }.
\end{split}
\end{equation*}

For the phenotype profile, we first apply a rank normal quantile tansformation to each of its dimensions suggested by [changshuai et. al.]. 
\begin{displaymath}
  q_{im}=\Phi^{-1}[(rank(y_{im})-0.5)/|Y|],
\end{displaymath} 
where $y_{im}$ is the value of the $m th.$ dimension of the target phenotype for the $i th.$ subjects and $|Y|$ is the dimensionality of the phenotype. Doing so will remove the complication due to the mixture of distribution types introduced by subcomponents of a high dimensional phenotype. As a result, the rank-normal-quantile transformed phenotype can be compared in a manner similar to that of vertex code:
\begin{equation} \label{eq_wSY}
\begin{split}
  S_{ij}^Y &= wED(q_{i.},q_{j.}) \\
  &=\exp
  {
    \Big[-\frac{\sum_{m=1}^{|Y|}{w_m(q_{im}-q_{jm})^2}} {\sum_{m=1}^{|Y|}{w_m}}\Big]
  },
\end{split}
\end{equation}
where $q_{im}$ and $q_{jm}$ are values of the $m th.$ dimension of the transformed phenotype for a pair of subjects; the weight $w_m$ is the ralative importance of the $m th.$ dimension of the phenotype. For a case control study with one dimensional phenotype, that is, $|Y|=|y|=1$, the above function simplifys to:
\begin{displaymath}
  S_{ij}^{y}=ED(q_i,q_j)=\exp{[-(q_i-q_j)^2]},
\end{displaymath}

Centralization of a similarity function is done by substracting each raw measurement $S_{ij}$ with the two conditional mean on subject i and j, then add the mean of all measurement to it [changshuai et. al.]. Takeing the weighted genetic similarity as a example, the centered measurement is:
\begin{displaymath}
  \tilde{S}_{ij}^{G}=S_{ij}^{G}-E(S_{i.}^{G})-E(S_{.j}^{G})+E(S_{..}^{G}).
\end{displaymath}
The same centralization scheme is applied to other similarity measurement, either weighted or unweighted.

The U statistics is the mean of the product of three centered similarity measurement, across all subject pairs:
\begin{displaymath}
  U^{GVY}=\frac{1}{N(N-1)}\sum_{i \neq j} \tilde{S}_{ij}^{G} \tilde{S}_{ij}^{V} \tilde{S}_{ij}^{Y},
\end{displaymath}
where N is the number of subjects, and $(i,j)$ indices a pair of subjects. 

For the actually implementation, three N by N symmetric similarity matrices are first constructed, and elementwisely multipled to get an u scores matrix, and the lower (or upper) traiangle elements are then summed up for the U statistics.

For now, the genetic and image similarity gauges are treated as U-weight terms, and the phenotype gauge is treated as U-kernel. In general, the assignment of U-kernal and U-weight terms does not alter the limiting distribution of the final U statistics, but fixing the U-kernel on phenotype has the added benifit of testing simpler hypothesis by dropping one of the U-weight terms.

%\begin{bibliography}
%\end{bibliography}

\end{document}
