\documentclass[12pt]{article}

\usepackage{geometry}
\geometry{a4paper, left=20mm, right=20mm, top=20mm, bottom=20mm}

\setlength{\parindent}{2em}
\setlength{\parskip}{1em}

\usepackage{hyperref}

\usepackage{syntonly}
%\syntaxonly
\usepackage{amsmath}
\usepackage{amssymb}

\usepackage[utf8]{inputenc}
\usepackage[style=numeric,sorting=none,]{biblatex}
\addbibresource{ref.bib}

\usepackage{graphicx}
\graphicspath{{img/}}

% aliasis
% FreeSurfer from Havord Unv.
\newcommand{\FS}{\href{http://surfer.nmr.mgh.harvard.edu}{\textbf{FreeSurfer}}} 

% encoders
% vector or matrix
\newcommand{\vecEC}[1]{\boldsymbol{#1}}

% decoders
\newcommand{\vecDC}[1]{\boldsymbol{\tilde{#1}}}

\newcommand{\xVO}{\boldsymbol{x}}         % the x vector, original
\newcommand{\xVR}{\boldsymbol{\tilde{x}}} % the x vector, recovered
\newcommand{\xSO}{x}                      % the x scaler, original
\newcommand{\xSR}{\tilde{x}}              % the x scaler, recovered

% the eta vector
\newcommand{\etaEC}{\vecEC{\eta}}                   % generic encoder
\newcommand{\etaEi}{\WEC_i^{d_{i+1} \times d_i}} % encoder layer i
\newcommand{\etaDC}{\vecDC{\eta}}                   % generic decoder
\newcommand{\etaDi}{\WDC_i^{d_i \times d_{i+1}}} % decoder layer #i

% the W matrix
\newcommand{\WEC}{\vecEC{W}}                   % generic encoder
\newcommand{\WEi}{\WEC_i^{d_{i+1} \times d_i}} % encoder layer i
\newcommand{\WDC}{\vecDC{W}}                   % generic decoder
\newcommand{\WDi}{\WDC_i^{d_i \times d_{i+1}}} % decoder layer #i

% the w vector
\newcommand{\wEC}{\vecEC{w}}                   % generic encoder
\newcommand{\wDC}{\vecDC{w}}                   % generic decoder

% the b vector
\newcommand{\bEC}{\vecEC{b}}             % generic encoder
\newcommand{\bEi}{\bEC_i^{d_i}}          % encoder layer i
\newcommand{\bDC}{\vecDC{b}}             % generic decoder
\newcommand{\bDi}{\bDC_i^{d_i}}          % encoder layer i

% the x vector
\newcommand{\xEC}{\vecEC{x}}             % generic encoder
\newcommand{\xDC}{\vecDC{x}}             % generic decoder

% the z vector
\newcommand{\zEC}{\vecEC{z}}           % generic encoder
\newcommand{\zDC}{\vecDC{z}}           % generic decoder

% I/O for decoder layer
\newcommand{\iDi}{\zDC_{i+1}^{d_{i+1}}}
\newcommand{\oDi}{\zDC_i^{d_i}}

% the vector of ones
\newcommand{\one}[1]{\boldsymbol{1}^{#1}}         % the z vector in encoders
% the diagnal matrix
\newcommand{\I}[1]{\boldsymbol{I}^{#1}}

% parameters in the neural network
\newcommand{\Par}{\boldsymbol{\Theta}} % the parameters
\newcommand{\pEC}{\boldsymbol{\theta}} % the parameters in the stacked autoencoder
\newcommand{\pDC}{\boldsymbol{\tilde{\theta}}} % the parameters in the decoder

% Loss function in Cross Entropy form
\newcommand{\LCE}[2]{#1\log{#2} + (1 - #1)\log{(1 - #2)}}

% partial derivative
\newcommand{\PDV}[2]{\frac{\partial #1}{\partial #2}}


\pagestyle{headings}

\author{Xiaoran Tong}

\begin{document}
\title{A non-parametric method for joint association analysis of
  sequencing and imaging data}
\maketitle

\begin{abstract}
  The next generation genome sequencing and neroimaging technology
  give rise to large, mult-site cohort with growing wealth of next
  generation sequencing (NGS) data and magnetic resonance imaging
  (MRI) data, which mandates the corresponding analytical methodology
  capable of utilizing both type of information to identify predictive
  biomarkers associated with complex disorders. Such attemp, however,
  are met with "the curse of dimensionality", due to the large number
  of variants in the genome and images. In this work, we tackled the
  dimensionality issue of 3D cortical surface vertices by training a
  deep encoder with the deep learning algorithm and generating a
  compact, abstracted representation of the original surface with the
  encoder per se. An U statistic with profile similarity based weight
  term were then adopted to evaluate the joint association of encoded
  surface and genome data with the phenotype. We showed by simulation
  that the method maintains the correct type 1 error rate, and
  achieved a statistical power higher then using either genome or
  image data alone, or using the original surface, or methods relying
  on large number of per-variant test. To illustrate our approach, we
  apply the proposed method to the genomic sequencing and neuroimage
  data from the Alzheimer's disease Neuroimaging Initiative (ADNI).
\end{abstract}

\input{tex/sec_intro}
%\input{tex/sec_mater}
\input{tex/sec_method}
%\input{tex/sec_result}
%\input{tex/sec_dissc}

\printbibheading
\printbibliography

\end{document}
