\documentclass[twocolumn]{article}

\usepackage{syntonly}
%\syntaxonly
\usepackage{amsmath}

\pagestyle{headings}

\author{Xiaoran Tong}

\begin{document}

\section{Simulation}
Simulations are based on 806 participants of ADNI study with both WGS and MRI data avaliable. Each iteration run choose a pair of testing units for genomc and image profiles. The genomic testing unit is a randomly gene randomly picked from the genomic features list supplied by GRC38. For neuroimage profile, the testing unit is regions of 512 vertices randomly located in the white matter (WM) surface reconstructed from MRI data. The genomic effect $\beta^G$ and vertex effect $\beta^G$ are then generated by randomly selecting 5\% of the variants (e.g. SNPs in a gene, vertices in a WM region) in both testing units and assigning a value drawn from $N(0,1)$. Given the row major genomic profile $X^G$ and image profile $X^V$ of all subjects, two basic continuous response $Y^G_C$ and $Y^V_C$ were additively calculated across variants in the corresponding testing unit; two additional continuous responses $Y^A_C$ and $Y^X_C$ were then created by adding up the the basics with or without an extra product term. Four binary responds $Y^G_b, Y^V_b, Y^A_b, Y^X_b$ were also generated by putting the four continuous ones through inverse logit and draw cases from the resulting probability. The 8 responses can be written as:
\begin{equation*} \label{eq:SIM}
\begin{split}
  \beta^G_k &= N(0,1) \times Bernoulli(0.05), k=1, \dots, |G| \\
  \beta^V_l &= N(0,1) \times Bernoulli(0.05), l=1, \dots, |V| \\
  \boldsymbol{Y^G_c} &= \boldsymbol{X^G \beta^G} \\
  \boldsymbol{Y^V_c} &= \boldsymbol{X^V \beta^V} \\
  \boldsymbol{Y^A_c} &= \boldsymbol{Y^G_c} + \boldsymbol{Y^V_c} \\
  \boldsymbol{Y^X_c} &= \boldsymbol{Y^G_c} + \boldsymbol{Y^V_c} + \boldsymbol{Y^G_c Y^V_c} \\
  \boldsymbol{Y^G_b} &= Bernolli(logit^{-1}(\boldsymbol{Y^G_c})) \\
  \boldsymbol{Y^V_b} &= Bernolli(logit^{-1}(\boldsymbol{Y^V_c})) \\
  \boldsymbol{Y^A_b} &= Bernolli(logit^{-1}(\boldsymbol{Y^A_c})) \\
  \boldsymbol{Y^X_b} &= Bernolli(logit^{-1}(\boldsymbol{Y^X_c}))
\end{split}
\end{equation*}
where $k$ and $l$ indices variants within the choson genomic and image units, and $|G|$ and $|V|$ is the variant count. For now $|V|$ is fixed to 512 -- the number of vertices in a WM sample region.

To see the power performance of the joint U-statistics and the benifit of vertex encoder, two factors are to be considered simutaniously in the simulation study. The first is construct of U statistics which may result in correct, partially correct or mis-specification when faced with the 8 generated responses. The available U-kernel options are genomic only (G), image only (V), or joint (G+V), corresponding to the 3 aforementioned hypothesis. The second factor is the construct of image kernel (V). Both the original and the encoded vertices can be used. Alternatively, as an analogy of GWAS, the widely used vertex-wise analysis (VWA) can also be applied. In brief, VWA first smoothes the vertices data via a gaussian blur process to reduce the noise ratio, then builds a weight kernel for each vertex and performs a large number of tests, after which the most significant one is chosen as the representative of the whole testing unit. All scensiable combinations of the two factors (e.g. VWA cannot be applied to encoded vertices) were tested on the 8 responses across a spectrium of simple sizes form 100 to 800.

As expected, U-kernel that exactly matches the response asertained the heighest power across all sample sizes, while a completely mis-specified U-kernel exhibit no power. A partially mis-specified kernel composition maintained a performance close to that of the correctly specified one, thus making the joint U-kernel (A+V) an overall optimal choice when prior knowledge of effect constituion is lacking [figure ?]. Limiting the comparesion among tests using raw vertices, it is observable that regional similarity U test could outperform the vertex-wise analysis (VWA) under all scenarios of effect and kernel composition. By looking at all similarity U tests involving vertices, the encoded vertices offer more power boost when sample size increase. When the kernel is correctly specified, the performance is always better when encoded vertices is used.

%\begin{bibliography}
%\end{bibliography}

\end{document}
